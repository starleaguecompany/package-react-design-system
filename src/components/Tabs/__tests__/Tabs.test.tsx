import * as React from 'react';
import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';

import { Tabs } from '../index';

describe('Tabs', () => {
  test('should render correctly with no props', () => {
    render(<Tabs />);

    expect(screen.getByTestId('Tabs')).toMatchSnapshot();
    expect(screen.getByTestId('Tabs')).not.toHaveClass('');
  });

  test('should render correctly with attributes', () => {
    render(<Tabs id="test-id" className="test-class" />);

    expect(screen.getByTestId('Tabs')).toMatchSnapshot();
    expect(screen.getByTestId('Tabs')).toHaveAttribute('id', 'test-id');
    expect(screen.getByTestId('Tabs')).toHaveClass('test-class');
  });

  test('should render correctly with tabs pain', () => {
    render(
      <Tabs>
        <Tabs.TabPane title="Pane 1">Pane content 1</Tabs.TabPane>
        <Tabs.TabPane title="Pane 2">Pane content 2</Tabs.TabPane>
      </Tabs>
    );

    expect(screen.getByTestId('Tabs')).toMatchSnapshot();
    expect(screen.getByText('Pane 1')).toBeVisible();
    expect(screen.getByText('Pane 2')).toBeVisible();
    expect(screen.getByText('Pane content 1')).toBeVisible();
    expect(screen.getByText('Pane content 2')).toBeVisible();
    expect(screen.getByText('Pane 1')).toHaveClass('active');
    expect(screen.getByText('Pane 2')).not.toHaveClass('active');
    expect(screen.getByText('Pane content 1')).toHaveClass('active');
    expect(screen.getByText('Pane content 2')).not.toHaveClass('active');
  });

  test('should render correctly with defaultIndex prop', () => {
    render(
      <Tabs defaultIndex={1}>
        <Tabs.TabPane title="Pane 1" />
        <Tabs.TabPane title="Pane 2" />
        <Tabs.TabPane title="Pane 3" />
      </Tabs>
    );

    expect(screen.getByTestId('Tabs')).toMatchSnapshot();
    expect(screen.getByText('Pane 1')).not.toHaveClass('active');
    expect(screen.getByText('Pane 2')).toHaveClass('active');
    expect(screen.getByText('Pane 3')).not.toHaveClass('active');
  });

  test('should execute onChange callback with tab pane index when click on it', () => {
    const onChangeCallback = jest.fn(tabPainIndex => tabPainIndex);
    render(
      <Tabs onChange={onChangeCallback}>
        <Tabs.TabPane title="Pane 1" />
        <Tabs.TabPane title="Pane 2" />
        <Tabs.TabPane title="Pane 3" />
      </Tabs>
    );

    expect(screen.getByTestId('Tabs')).toMatchSnapshot();
    userEvent.click(screen.getByText('Pane 3'));
    expect(onChangeCallback).toBeCalledTimes(1);
    expect(onChangeCallback).toReturnWith(2);
  });
});
