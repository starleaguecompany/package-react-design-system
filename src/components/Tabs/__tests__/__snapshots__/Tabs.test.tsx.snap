// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tabs should execute onChange callback with tab pane index when click on it 1`] = `
<div
  data-qa="Tabs"
>
  <div
    class="tabs container direction-horizontal align-center"
    data-qa="Space"
    style="column-gap: 20px; row-gap: 20px;"
  >
    <div
      class="tab active container direction-horizontal align-center inline"
      data-index="0"
      data-qa="Space"
      style="column-gap: 4px; row-gap: 4px;"
    >
      Pane 1
    </div>
    <div
      class="tab container direction-horizontal align-center inline"
      data-index="1"
      data-qa="Space"
      style="column-gap: 4px; row-gap: 4px;"
    >
      Pane 2
    </div>
    <div
      class="tab container direction-horizontal align-center inline"
      data-index="2"
      data-qa="Space"
      style="column-gap: 4px; row-gap: 4px;"
    >
      Pane 3
    </div>
  </div>
  <div
    class="panes"
  >
    <div
      class="pane active pane"
      data-qa="TabPane"
      title="Pane 1"
    />
    <div
      class="pane pane"
      data-qa="TabPane"
      title="Pane 2"
    />
    <div
      class="pane pane"
      data-qa="TabPane"
      title="Pane 3"
    />
  </div>
</div>
`;

exports[`Tabs should render correctly with attributes 1`] = `
<div
  class="test-class"
  data-qa="Tabs"
  id="test-id"
>
  <div
    class="tabs container direction-horizontal align-center"
    data-qa="Space"
    style="column-gap: 20px; row-gap: 20px;"
  />
  <div
    class="panes"
  />
</div>
`;

exports[`Tabs should render correctly with defaultIndex prop 1`] = `
<div
  data-qa="Tabs"
>
  <div
    class="tabs container direction-horizontal align-center"
    data-qa="Space"
    style="column-gap: 20px; row-gap: 20px;"
  >
    <div
      class="tab container direction-horizontal align-center inline"
      data-index="0"
      data-qa="Space"
      style="column-gap: 4px; row-gap: 4px;"
    >
      Pane 1
    </div>
    <div
      class="tab active container direction-horizontal align-center inline"
      data-index="1"
      data-qa="Space"
      style="column-gap: 4px; row-gap: 4px;"
    >
      Pane 2
    </div>
    <div
      class="tab container direction-horizontal align-center inline"
      data-index="2"
      data-qa="Space"
      style="column-gap: 4px; row-gap: 4px;"
    >
      Pane 3
    </div>
  </div>
  <div
    class="panes"
  >
    <div
      class="pane pane"
      data-qa="TabPane"
      title="Pane 1"
    />
    <div
      class="pane active pane"
      data-qa="TabPane"
      title="Pane 2"
    />
    <div
      class="pane pane"
      data-qa="TabPane"
      title="Pane 3"
    />
  </div>
</div>
`;

exports[`Tabs should render correctly with no props 1`] = `
<div
  data-qa="Tabs"
>
  <div
    class="tabs container direction-horizontal align-center"
    data-qa="Space"
    style="column-gap: 20px; row-gap: 20px;"
  />
  <div
    class="panes"
  />
</div>
`;

exports[`Tabs should render correctly with tabs pain 1`] = `
<div
  data-qa="Tabs"
>
  <div
    class="tabs container direction-horizontal align-center"
    data-qa="Space"
    style="column-gap: 20px; row-gap: 20px;"
  >
    <div
      class="tab active container direction-horizontal align-center inline"
      data-index="0"
      data-qa="Space"
      style="column-gap: 4px; row-gap: 4px;"
    >
      Pane 1
    </div>
    <div
      class="tab container direction-horizontal align-center inline"
      data-index="1"
      data-qa="Space"
      style="column-gap: 4px; row-gap: 4px;"
    >
      Pane 2
    </div>
  </div>
  <div
    class="panes"
  >
    <div
      class="pane active pane"
      data-qa="TabPane"
      title="Pane 1"
    >
      Pane content 1
    </div>
    <div
      class="pane pane"
      data-qa="TabPane"
      title="Pane 2"
    >
      Pane content 2
    </div>
  </div>
</div>
`;
